src/Compilers/CSharp/Portable/Errors/LazyNullableContraintChecksDiagnosticInfo.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// A lazily calculated diagnostic for constraint check nullability.
```

Comment:
>nullability [](start = 60, length = 11)

There is a hidden contract in this implementation: that the only warnings produced during `ConstraintsHelper.CheckTypeConstraints` are nullability warnings. That contract should be documented at least.

Reply:
The core `CheckConstraints` parameter is documented and I've added a similar comment here.

---
In reply to: [209389534](https://github.com/dotnet/roslyn/pull/29197#discussion_r209389534) [](ancestors = 209389534)